### ISSUE TITLE
Dom object leak with tooltips

### ISSUE THREAD

**Issue by KenMills** [GwtMaterialDesign/gwt-material#494]
If I create a button dynamically, add a tooltip using button.setTooltip("tooltip"), and add it to a panel.

Whenever I leave the page with the button a Dom tooltip element is created. When I return to the page another DOM element is created.

In the gif I created, the "to test" was created dynamically.
The "go home" button was created in the uibinder xml.

Whenever I go back home the tooltip count in the DOM is increased.

In my real app I have about 50 buttons with tooltips on a page (most are hidden, but the DOM elements still show up. In a short time there will be 1,000s of unused tooltip objects.

![tooltip](https://cloud.githubusercontent.com/assets/12837618/20111939/0b1dea38-a59f-11e6-94b2-4a60af088086.gif)

**Comment by KenMills** [comment]
In case it's needed, here's the sample app I used.
[TooltipTest.zip](https://github.com/GwtMaterialDesign/gwt-material/files/579062/TooltipTest.zip)

**Comment by kevzlou7979** [comment]
Thanks for cathing this up, will take a look on it.

**Comment by BenDol** [comment]
What version are you using?

**Comment by BenDol** [comment]
In the demo you are using 1.5.2, can you test it with 2.0-SNAPSHOT? I am trying to reproduce the issue without success. Thanks.

**Comment by KenMills** [comment]
I'm currently running version 1.5.2 (I've tried running 2.0, but am running into compile issues and can't get v2 to build)

**Comment by kevzlou7979** [comment]
Can you show us the compile log issues while migrating to 2.0-SNAPSHOT.

**Comment by KenMills** [comment]
Attaching my log files.
1.5.2 builds fine.
I see errors when changing to 2.0-SNAPSHOT in my pom.xml (the only change made).

Are there other changes required?
[build.zip](https://github.com/GwtMaterialDesign/gwt-material/files/580050/build.zip)

**Comment by BenDol** [comment]
Change:
```xml
<m:MaterialButton ui:field="btnHome" text="To Home" tooltip="Go Home" backgroundColor="blue"/>
```
to:
```xml
<m:MaterialButton ui:field="btnHome" text="To Home" tooltip="Go Home" backgroundColor="BLUE"/>
```

**Comment by KenMills** [comment]
I get the same error when changing the color from "blue" to "BLUE".


from log:
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building TooltipTest 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ TooltipTest ---
[INFO] Deleting C:\Users\kenm\workspace\GWTP-Test\TooltipTest\target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ TooltipTest ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Users\kenm\workspace\GWTP-Test\TooltipTest\src\main\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.2:compile (default-compile) @ TooltipTest ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 16 source files to C:\Users\kenm\workspace\GWTP-Test\TooltipTest\target\TooltipTest-1.0-SNAPSHOT\WEB-INF\classes
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ TooltipTest ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Users\kenm\workspace\GWTP-Test\TooltipTest\src\test\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.2:testCompile (default-testCompile) @ TooltipTest ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to C:\Users\kenm\workspace\GWTP-Test\TooltipTest\target\test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.17:test (default-test) @ TooltipTest ---
[INFO] Surefire report directory: C:\Users\kenm\workspace\GWTP-Test\TooltipTest\target\surefire-reports

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.example.myproject.client.SandboxJukitoTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.738 sec - in com.example.myproject.client.SandboxJukitoTest

Results :

Tests run: 1, Failures: 0, Errors: 0, Skipped: 0

[INFO] 
[INFO] --- gwt-maven-plugin:2.7.0:compile (default) @ TooltipTest ---
[ERROR] Nov 08, 2016 11:09:01 PM java.util.prefs.WindowsPreferences <init>
[ERROR] WARNING: Could not open/create prefs root node Software\JavaSoft\Prefs at root 0x80000002. Windows RegCreateKeyEx(...) returned error code 5.
[INFO] Compiling module com.example.myproject.TooltipTest
[INFO]    Ignored 149 units with compilation errors in first pass.
[INFO] Compile with -strict or with -logLevel set to TRACE or DEBUG to see all errors.
[INFO]    Computing all possible rebind results for 'com.gwtplatform.mvp.client.DesktopGinjector'
[INFO]       Rebinding com.gwtplatform.mvp.client.DesktopGinjector
[INFO]          Invoking generator com.google.gwt.inject.rebind.GinjectorGenerator
[INFO]             [ERROR] Error injecting com.example.myproject.client.application.testpage.TestPageView$Binder: Unable to create or inherit binding: No @Inject or default constructor found for com.example.myproject.client.application.testpage.TestPageView$Binder
[INFO]   Path to required node:
[INFO] 
[INFO] com.example.myproject.client.application.testpage.TestPageView [com.gwtplatform.mvp.client.gin.AbstractPresenterModule.bindPresenter(AbstractPresenterModule.java:120)]
[INFO]  -> com.example.myproject.client.application.testpage.TestPageView$Binder [@Inject constructor of com.example.myproject.client.application.testpage.TestPageView]
[INFO] 
[INFO]    [ERROR] Errors in 'gen/com/gwtplatform/mvp/client/DesktopGinjectorProvider.java'
[INFO]       [ERROR] Line 8: Failed to resolve 'com.gwtplatform.mvp.client.DesktopGinjector' via deferred binding
[INFO]    [WARN] For the following type(s), generated source was never committed (did you forget to call commit()?)
[INFO]       [WARN] com.gwtplatform.mvp.client.com_gwtplatform_mvp_client_DesktopGinjectorImpl
[ERROR] Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=256M; support was removed in 8.0
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 40.859 s
[INFO] Finished at: 2016-11-08T23:09:27-08:00
[INFO] Final Memory: 30M/322M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:gwt-maven-plugin:2.7.0:compile (default) on project TooltipTest: Command [[
[ERROR] C:\Program Files\Java\jdk1.8.0_101\jre\bin\java -Xss1024K -Xmx1024M -XX:MaxPermSize=256M -classpath C:\Users\kenm\workspace\GWTP-Test\TooltipTest\target\TooltipTest-1.0-SNAPSHOT\WEB-INF\classes;C:\Users\kenm\workspace\GWTP-Test\TooltipTest\src\main\java;C:\Users\kenm\workspace\GWTP-Test\TooltipTest\target\generated-sources\annotations;C:\Users\kenm\.m2\repository\com\google\gwt\gwt-user\2.7.0\gwt-user-2.7.0.jar;C:\Users\kenm\.m2\repository\javax\validation\validation-api\1.0.0.GA\validation-api-1.0.0.GA.jar;C:\Users\kenm\.m2\repository\javax\validation\validation-api\1.0.0.GA\validation-api-1.0.0.GA-sources.jar;C:\Users\kenm\.m2\repository\com\gwtplatform\gwtp-mvp-client\1.5.3\gwtp-mvp-client-1.5.3.jar;C:\Users\kenm\.m2\repository\com\gwtplatform\gwtp-clients-common\1.5.3\gwtp-clients-common-1.5.3.jar;C:\Users\kenm\.m2\repository\com\gwtplatform\gwtp-mvp-shared\1.5.3\gwtp-mvp-shared-1.5.3.jar;C:\Users\kenm\.m2\repository\com\google\gwt\gwt-dev\2.7.0\gwt-dev-2.7.0.jar;C:\Users\kenm\.m2\repository\org\ow2\asm\asm\5.0.3\asm-5.0.3.jar;C:\Users\kenm\.m2\repository\org\ow2\asm\asm-util\5.0.3\asm-util-5.0.3.jar;C:\Users\kenm\.m2\repository\org\ow2\asm\asm-tree\5.0.3\asm-tree-5.0.3.jar;C:\Users\kenm\.m2\repository\org\ow2\asm\asm-commons\5.0.3\asm-commons-5.0.3.jar;C:\Users\kenm\.m2\repository\org\apache\velocity\velocity\1.7\velocity-1.7.jar;C:\Users\kenm\.m2\repository\commons-collections\commons-collections\3.2.1\commons-collections-3.2.1.jar;C:\Users\kenm\.m2\repository\commons-lang\commons-lang\2.4\commons-lang-2.4.jar;C:\Users\kenm\.m2\repository\com\gwtplatform\gwtp-dispatch-rpc-client\1.5.3\gwtp-dispatch-rpc-client-1.5.3.jar;C:\Users\kenm\.m2\repository\com\gwtplatform\gwtp-dispatch-common-client\1.5.3\gwtp-dispatch-common-client-1.5.3.jar;C:\Users\kenm\.m2\repository\com\gwtplatform\gwtp-dispatch-rpc-server-guice\1.5.3\gwtp-dispatch-rpc-server-guice-1.5.3.jar;C:\Users\kenm\.m2\repository\com\gwtplatform\gwtp-dispatch-rpc-server\1.5.3\gwtp-dispatch-rpc-server-1.5.3.jar;C:\Users\kenm\.m2\repository\com\gwtplatform\gwtp-dispatch-rpc-shared\1.5.3\gwtp-dispatch-rpc-shared-1.5.3.jar;C:\Users\kenm\.m2\repository\com\gwtplatform\gwtp-dispatch-common-shared\1.5.3\gwtp-dispatch-common-shared-1.5.3.jar;C:\Users\kenm\.m2\repository\com\google\inject\guice\3.0\guice-3.0.jar;C:\Users\kenm\.m2\repository\javax\inject\javax.inject\1\javax.inject-1.jar;C:\Users\kenm\.m2\repository\aopalliance\aopalliance\1.0\aopalliance-1.0.jar;C:\Users\kenm\.m2\repository\com\google\inject\extensions\guice-servlet\3.0\guice-servlet-3.0.jar;C:\Users\kenm\.m2\repository\com\google\inject\extensions\guice-assistedinject\3.0\guice-assistedinject-3.0.jar;C:\Users\kenm\.m2\repository\com\google\gwt\inject\gin\2.1.2\gin-2.1.2.jar;C:\Users\kenm\.m2\repository\com\github\gwtmaterialdesign\gwt-material\2.0-SNAPSHOT\gwt-material-2.0-SNAPSHOT.jar;C:\Users\kenm\.m2\repository\com\github\gwtmaterialdesign\gwt-material-jquery\1.0-SNAPSHOT\gwt-material-jquery-1.0-SNAPSHOT.jar;C:\Users\kenm\.m2\repository\com\github\gwtmaterialdesign\gwt-material-themes\2.0-SNAPSHOT\gwt-material-themes-2.0-SNAPSHOT.jar;C:\Users\kenm\.m2\repository\com\github\gwtmaterialdesign\gwt-material-addins\2.0-SNAPSHOT\gwt-material-addins-2.0-SNAPSHOT.jar;C:\Users\kenm\.m2\repository\com\google\gwt\gwt-user\2.7.0\gwt-user-2.7.0.jar;C:\Users\kenm\.m2\repository\javax\validation\validation-api\1.0.0.GA\validation-api-1.0.0.GA.jar;C:\Users\kenm\.m2\repository\javax\validation\validation-api\1.0.0.GA\validation-api-1.0.0.GA-sources.jar;C:\Users\kenm\.m2\repository\com\google\gwt\gwt-dev\2.7.0\gwt-dev-2.7.0.jar;C:\Users\kenm\.m2\repository\org\ow2\asm\asm\5.0.3\asm-5.0.3.jar;C:\Users\kenm\.m2\repository\org\ow2\asm\asm-util\5.0.3\asm-util-5.0.3.jar;C:\Users\kenm\.m2\repository\org\ow2\asm\asm-tree\5.0.3\asm-tree-5.0.3.jar;C:\Users\kenm\.m2\repository\org\ow2\asm\asm-commons\5.0.3\asm-commons-5.0.3.jar com.google.gwt.dev.Compiler -logLevel INFO -style OBF -war C:\Users\kenm\workspace\GWTP-Test\TooltipTest\target\TooltipTest-1.0-SNAPSHOT -localWorkers 8 -XfragmentCount -1 -sourceLevel auto -gen C:\Users\kenm\workspace\GWTP-Test\TooltipTest\target\.generated com.example.myproject.TooltipTest
[ERROR] ]] failed with status 1
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

**Comment by KenMills** [comment]
Do you see the error in 1.5.2? 
If it's fixed in the latest, great. I'll have to work on getting my app up to java 1.8  since we have other components that only work at 1.7...

**Comment by kevzlou7979** [comment]
Might be good if you update your project to java 1.8.

**Comment by KenMills** [comment]
That's going to take some time... a few of our libraries aren't compatible with 1.8...

I still can't get the simple test app to compile using 2.0 snapshot, but it builds/runs fine in 1.5.2.
Any ideas?

I was able to build and run the 2.0 snapshot demo.
I modified it and was able to repro the issue there as well.

I added a panel to the Dialogs view and created a button with a tooltip.
[DialogsView.zip](https://github.com/GwtMaterialDesign/gwt-material/files/582350/DialogsView.zip)

![tooltip2](https://cloud.githubusercontent.com/assets/12837618/20161505/c93ef70a-a6a3-11e6-9483-782912ae8110.gif)
